src/Streamly/FileSystem/FDIO.hs

```diff
@@ -22,6 +22,9 @@ module Streamly.FileSystem.FDIO
     )
 where
 
+#if defined(mingw32_HOST_OS)
+import Control.Monad (when)
```

Comment:
Since we need this import in all cases we can take it out of the CPP conditional.